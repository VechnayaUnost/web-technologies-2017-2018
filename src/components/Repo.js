import React from 'react';

class Repo extends React.Component {
    render() {
        const repo = this.props.repo;
        return (
            <li className="repo">
                <div className="repo-description">
                    <h3>{repo.name}</h3>
                    <p>{repo.description}</p>
                </div>
                <div className="repo-stars-lang">
                    {
                        repo.language
                        ?
                        <span className="repo-lang">Language: {repo.language}</span>
                        :
                        null
                    }
                    <span className="repo-stars">Stars: {repo.stargazers_count}</span>
                </div>
            </li>
        )
    }
}

export default Repo;