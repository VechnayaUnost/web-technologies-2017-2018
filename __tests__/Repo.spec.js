import React from 'react';
import Repo from '../src/components/Repo';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Repo component test', () => {
    it('Repo component rendered succesfully', () => {
        const props = {
            repo: {
                name: 'AvatarSkeleton',
                description: 'Creates Avatar skeleton with some preinstalled options',
                stargazers_count: 321,
                language: 'C#',
            }
        }

        const repo = mount(
            <Repo {...props} />
        );
        expect(mountToJson(repo)).toMatchSnapshot();
    });

    it('Repo component rendered widhout language', () => {
        const props = {
            repo: {
                name: 'AvatarSkeleton',
                description: 'Creates Avatar skeleton with some preinstalled options',
                stargazers_count: 321,
                language: '',
            }
        }

        const repo = mount(
            <Repo {...props} />
        );
        expect(repo.find('.repo-lang').length).toBe(0);
    });
});