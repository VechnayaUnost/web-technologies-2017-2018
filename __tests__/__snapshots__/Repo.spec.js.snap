// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo component test Repo component rendered succesfully 1`] = `
<Repo
  repo={
    Object {
      "description": "Creates Avatar skeleton with some preinstalled options",
      "language": "C#",
      "name": "AvatarSkeleton",
      "stargazers_count": 321,
    }
  }
>
  <li
    className="repo"
  >
    <div
      className="repo-description"
    >
      <h3>
        AvatarSkeleton
      </h3>
      <p>
        Creates Avatar skeleton with some preinstalled options
      </p>
    </div>
    <div
      className="repo-stars-lang"
    >
      <span
        className="repo-lang"
      >
        Language: 
        C#
      </span>
      <span
        className="repo-stars"
      >
        Stars: 
        321
      </span>
    </div>
  </li>
</Repo>
`;
